src/Corax/Queries/SortingMatch.Comparers.cs

```diff
@@ -0,0 +1,305 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Corax.Queries
+{
+    public enum MatchCompareFieldType
+    {
+        Sequence,
+        Integer,
+        Floating
+    }
+
+    public interface IMatchComparer
+    {        
+        MatchCompareFieldType FieldType { get; }
+
+        int FieldId { get; }
+
+        int CompareById(long idx, long idy);
+
+        int CompareSequence(ReadOnlySpan<byte> sx, ReadOnlySpan<byte> sy);
+        int CompareNumerical<T>(T sx, T sy) where T : unmanaged;
+    }
+
+    partial struct SortingMatch
+    {                     
+        private static class BasicComparers
+        {
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            internal static int CompareAscending(ReadOnlySpan<byte> x, ReadOnlySpan<byte> y)
+            {
+                return x.SequenceCompareTo(y);
+            }
+
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            internal static int CompareAscending<T>(T x, T y)
+            {
+                if (typeof(T) == typeof(long))
+                {
+                    return Math.Sign((long)(object)y - (long)(object)x);
+                }
+                else if (typeof(T) == typeof(double))
+                {
+                    return Math.Sign((double)(object)y - (double)(object)x);
```

Comment:
Why not call `double.CompareTo`, etc?

Reply:
Good point, will see if it generated the proper code.